compiler/optimizations/gpuTransforms.cpp

```diff
@@ -230,6 +230,13 @@ std::queue<FnSymbol*> CreateGpuFunctionSpecializations::findAndCloneOnFns() {
 }
 
 void CreateGpuFunctionSpecializations::findAndCloneFnsReachableFromQueue(std::queue<FnSymbol*> &queue) {
+  // We rewrite wrap_on functions (those flagged with FLAG_ON_BLOCK) to dispatch
+  // to the either a gpu or cpu specialized "on" function, we don't need to
+  // clone these functions themselves.
+  auto shouldExemptCallToFn = [](FnSymbol *fn) {
+    return fn->hasFlag(FLAG_ON_BLOCK);
+  };
```

Comment:
I remember during a screenshare session we were talking about whether `wrap_on`s themselves should be cloned. Can you remind me what that was about? I think it had to do with wide references, but what we were talking about was probably just something nice to have rather than fixing an issue. But I can't be sure.

Reply:
I think the thought was:

- Things that we assign wide-pointers need to become wide themselves
- we pass the `wrap_on` function a kind of "package" of arguments that we forward to either the cpu-or-gpu specialized 'on' function. As part of that passing process the package itself becomes a wide pointer.
- So we thought maybe we could avoid that by moving things back a level and having the dispatch to cpu v. gpu specializations occur before we do the packaging

But then, thinking about that some more, that doesn't really resolve the whole "things that we assign to wide-pointers become wide themselves" thing so we gave up that idea for the time being.

So maybe if we strengthen how the wide-pointer transform works it might be advantageous to clone the `wrap_on` functions and move the gpu-or-cpu-specialization-dispatch back a level. But I don't think we're really very certain of that.

Reply:
OK thanks, yeah that was it. Just as an off the cuff remark, we can consider narrowing wide references before entering a GPU path, too.